File: content/search/source/includes/view/code-snippets/nodejs/add-modify-fields-create-index.js
import { MongoClient } from "mongodb";

// connect to your Atlas deployment
const uri =  "<connection-string>";

const client = new MongoClient(uri);

async function run() {
  try {
    const database = client.db("sample_airbnb");
    const collection = database.collection("listingsAndReviews_totalPrice");
    
    // define your MongoDB Search index
    const index = {
        name: "totalPriceIndex",
        definition: {
          "mappings": {
            "dynamic": true
          },
          "storedSource": {
            "include": [
              "totalPrice"
            ]
          }
        }
    }

    // run the helper method
    const result = await collection.createSearchIndex(index);
    console.log(result);
  } finally {
    await client.close();
  }
}
run().catch(console.dir);

File: content/search/source/includes/view/code-snippets/csharp/AddModifyFieldsCreateIndex.cs
using MongoDB.Bson;
using MongoDB.Driver;

// Connect to your Atlas deployment
var uri = "<connectionString>";
var client = new MongoClient(uri);
var database = client.GetDatabase("sample_airbnb");
var collection = database.GetCollection<BsonDocument>(
    "listingsAndReviews_totalPrice");

// Define your MongoDB Search index
var index = new CreateSearchIndexModel(
  "totalPriceIndex", new BsonDocument
  {
    { "mappings", new BsonDocument
      {
        { "dynamic", true }
      }
    },
    { "storedSource", new BsonDocument
      {
        { "include", new BsonArray { "totalPrice" } }
      }
    }
  });

// Create the index
var result = collection.SearchIndexes.CreateOne(index);
Console.WriteLine($"New index name: {result}");
